database_cleanup/tests/test_purge_modules.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import tagged
+
+from .common import Common, environment
+
+
+# Use post_install to get all models loaded more info: odoo/odoo#13458
+@tagged("post_install", "-at_install")
+class TestCleanupPurgeLineModule(Common):
+    def setUp(self):
+        super().setUp()
+        with environment() as env:
+            # create a nonexistent module
+            self.module = env["ir.module.module"].create(
+                {
+                    "name": "database_cleanup_test",
+                    "state": "to upgrade",
+                }
+            )
```

Comment:
There is a problem with the test case here. The data generated from the **environment()** method is being saved directly to the database. And when the test case ends, it is not being rolled back. Let's rollback it. Hint: you can use **tearDownClass()**

Reply:
![image](https://github.com/user-attachments/assets/288e6fee-f985-4953-9d52-53647f153454)